src/vos/vos_gc.c

```diff
@@ -739,6 +739,16 @@ vos_gc_run(int *credits)
 		return -DER_INVAL;
 	}
 
+	if (d_list_empty(pools)) {
+		/* Garbage collection has nothing to do.  Just consume the
+		 * credits and return to check when more credits are available.
+		 * No logging here as it causes log explosion when trace is
+		 * set.
+		 */
+		*credits = 0;
```

Comment:
I'm not sure if we should do this because caller might try to call this function again if we consumed all credits. If we just return all credits, then caller will think there is nothing to do and break the loop